lib/Onyx.js

```diff
@@ -780,6 +780,39 @@ function mergeCollection(collectionKey, collection) {
         });
 }
 
+/**
+ * Insert API responses and lifecycle data into Onyx
+ *
+ * @param {Object} data An object containing the Onyx data to set
+ * @param {String} data.onyxMethod One of [set, merge] Onyx methods
+ * @param {String} data.key ONYXKEY to set
+ * @param {*} data.value Value to store in ONYXKEY
+ */
+function update(data) {
+    // First, validate the Onyx object is in the format we expect
+    _.each(data, ({onyxMethod, key, value}) => {
+        if (!_.contains(['set', 'merge'], onyxMethod)) {
+            throw new Error(`Invalid onyxMethod ${onyxMethod} in Onyx update.`);
+        }
+        if (!_.isString(key)) {
+            throw new Error(`Invalid ${typeof key} key provided in Onyx update. Onyx key must be of type string.`);
+        }
+        if (typeof value === 'undefined') {
+            throw new Error(`Undefined value passed to key ${key} in Onyx update.`);
+        }
+    });
+
+    _.each(data, ({onyxMethod, key, value}) => {
+        if (onyxMethod === 'set') {
+            set(key, value);
```

Comment:
NAB: It might be nice to collect all the things using set and pass them to `multiSet()`, but since we are trying to move away from using `set()` at all, the could be a preoptimization. I kinda wish we had a `multiMerge()` though.

Reply:
I'm not sure how much of a pre-optimization that is, so I'll skip this for now.